src/main/java/ch/njol/skript/effects/EffReplace.java

```diff
@@ -41,98 +41,118 @@
-@Since("2.0, 2.2-dev24 (replace in multiple strings and replace items in inventory), 2.5 (replace first, case sensitivity)")
+@Description("Replaces all occurrences of a given text/regex with another text. Please note that you can only change " +
+	"variables and a few expressions, e.g. a <a href='/expressions.html#ExprMessage'>message</a> or a line of a sign.")
+@Examples({
+	"replace \"<item>\" in {_msg} with \"[%name of player's tool%]\"",
+	"replace every \"&\" with \"§\" in line 1 of targeted block",
+	"",
+	"# Very simple chat censor",
+	"on chat:",
+	"\treplace all \"kys\", \"idiot\" and \"noob\" with \"****\" in the message",
+	"\treplace using regex \"\\b(kys|idiot|noob)\\b\" with \"****\" in the message # Regex version for better results",
+	"",
+	"replace all stone and dirt in player's inventory and player's top inventory with diamond"
+})
+@Since("2.0, 2.2-dev24 (multiple strings, items in inventory), 2.5 (replace first, case sensitivity)," +
+	"INSERT VERSION (regex)")
 public class EffReplace extends Effect {
 
 	static {
 		Skript.registerEffect(EffReplace.class,
-				"replace (all|every|) %strings% in %strings% with %string% [(1¦with case sensitivity)]",
-				"replace (all|every|) %strings% with %string% in %strings% [(1¦with case sensitivity)]",
-				"replace first %strings% in %strings% with %string% [(1¦with case sensitivity)]",
-				"replace first %strings% with %string% in %string% [(1¦with case sensitivity)]",
+				"replace (all:(all|every)|:first|) %strings% in %strings% with %string% [(case:with case sensitivity)]",
+				"replace (all:(all|every)|:first|) %strings% with %string% in %strings% [(case:with case sensitivity)]",
+				"(regex:(replace [using] regex|regex replace)) %strings% in %strings% with %string%",
+				"(regex:(replace [using] regex|regex replace)) %strings% with %string% in %strings%",
 				"replace (all|every|) %itemtypes% in %inventories% with %itemtype%",
 				"replace (all|every|) %itemtypes% with %itemtype% in %inventories%");
 	}
-	
+
 	@SuppressWarnings("null")
 	private Expression<?> haystack, needles, replacement;
-	private boolean replaceString = true;
-	private boolean replaceFirst = false;
+	private boolean replaceString;
+	private boolean replaceRegex;
+	private boolean replaceItems;
+	private boolean replaceFirst;
 	private boolean caseSensitive = false;
 
-	@SuppressWarnings({"null"})
 	@Override
+	@SuppressWarnings("null")
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		haystack =  exprs[1 + matchedPattern % 2];
-		replaceString = matchedPattern < 4;
-		replaceFirst = matchedPattern > 1 && matchedPattern < 4;
+		replaceString = matchedPattern < 2;
+		replaceFirst = parseResult.hasTag("first");
+		replaceRegex = parseResult.hasTag("regex");
+		replaceItems = matchedPattern == 4 || matchedPattern == 5;
 		if (replaceString && !ChangerUtils.acceptsChange(haystack, ChangeMode.SET, String.class)) {
```

Comment:
This check is now only performed for the first 2 patterns, but should also be done for the regex replace. I recommend you set replaceString to `matchedPattern < 4`, line 107 can then be set to `if (replaceString) {`